### PR TITLE
Share ext error

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
This PR wraps the share extension components in an error boundary which allows crashes to be caught by `componentDidCatch`. The caught errors are sent to sentry and an error message is displayed allowing the user to close the extension gracefully.

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file updates

#### Screenshots
screenshot_1523637129
simulator screen shot - iphone 8 - 2018-04-13 at 09 38 22

@jarredwitt

### CODE REVIEW

**Comment by enahum** [comment]
@jarredwitt eslint errors

File: .eslintrc.json [hunk 1]
```diff
@@ -10,10 +10,7 @@
         }
     },
     "parser": "babel-eslint",
-    "plugins": [
-        "react",
-        "mocha"
-    ],
+    "plugins": ["react", "mocha"],
```
Comment (enahum) [inline .eslintrc.json]: why the changes on this file? do we really need them?

File: share_extension/android/extension.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import {captureException, initializeSentry, LOGGER_EXTENSION} from 'app/utils/sentry';
+
+import Navigation from './navigation';
+
+const ShareExtension = NativeModules.MattermostShare;
+
+export default class ShareApp extends PureComponent {
+    static contextTypes = {
+      intl: intlShape,
+      store: PropTypes.object.isRequired
```
Comment (enahum) [inline share_extension/android/extension.js]: do we need this? I think is not being used

File: share_extension/android/extension.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    constructor() {
+        super();
+        initializeSentry();
+    }
+
+    componentDidCatch(error) {
+        const {intl, store} = this.context;
+        const {formatMessage} = intl;
+
+        // captureException(error, LOGGER_EXTENSION, store);
```
Comment (enahum) [inline share_extension/android/extension.js]: what about this?

File: share_extension/ios/error_message.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    const {close, formatMessage} = props;
+    const theme = Preferences.THEMES.default;
+    const styles = getStyleSheet(theme);
+
+    return (
+        <View style={styles.errorWrapper}>
+            <View style={styles.errorContainer}>
+                <View style={styles.errorContent}>
+                    <View style={styles.errorMessage}>
+                        <Text style={styles.errorMessageText}>
```
Comment (enahum) [inline share_extension/ios/error_message.js]: we use here `FormattedText` instead

File: share_extension/ios/error_message.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+                            {formatMessage({
+                                id: 'mobile.share_extension.error_message',
+                                defaultMessage: 'An error has occurred while using the share extension.',
+                            })}
+                        </Text>
+                    </View>
+                    <TouchableOpacity
+                        style={styles.errorButton}
+                        onPress={() => close()}
```
Comment (enahum) [inline share_extension/ios/error_message.js]: change this to `onPress={close}`

File: share_extension/ios/error_message.js [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+                    >
+                        <Text style={styles.errorButtonText}>
```
Comment (enahum) [inline share_extension/ios/error_message.js]: same as above

File: .eslintrc.json (hunk 1 above)
Comment (jarredwitt) [reply .eslintrc.json]: Looks like my editor formatted it. I'll revert

File: share_extension/ios/error_message.js (hunk 5 above)
Comment (jarredwitt) [reply share_extension/ios/error_message.js]: `close` is expecting a data arg which is why there is an arrow in the properties. Without the arrow in the properties an event object would get sent and that messes up the `ShareExtension.close(data, Config.AppGroupId);` since it thinks it has a valid data object. Would you prefer another function be created that doesn't take a data arg? I'm good either way, just used an arrow since these components aren't going to be re-rendered due to state changes.

File: share_extension/ios/error_message.js (hunk 6 above)
Comment (jarredwitt) [reply share_extension/ios/error_message.js]: Same as above
